### PR TITLE
[Fixes #4085] Controller helper inconsistency: HttpNotFound(), Ok()

### LINKED ISSUE #4085: Controller helper inconsistency: HttpNotFound(), Ok()
I can `return HttpNotFound()` or I can `return Ok()` from a Controller. Why the inconsistent naming?

It should be one of these two alternatives IMO:

*  `return HttpNotFound()` and `return HttpOk()`, or
*  `return NotFound()` and `return Ok()`

I personally prefer the latter like the previous MVC incarnations. Thoughts folks?
cc @JakeGinnivan

### PR DESCRIPTION
**Pull request by kichalla**
@Eilon

### CODE REVIEW

**Comment by javiercn** [comment]
@kichalla A lot of these ones were inconsistent to maintain compatibility with MVC, at the time I remember that the decision was to keep compatibility vs consistency. I personally like this change, but just wanted to make sure this was taken in consideration

**Comment by rynowak** [comment]
It was.

**Comment by kichalla** [comment]
@javiercn More details here: https://github.com/aspnet/Mvc/issues/4085

File: test/Microsoft.AspNetCore.Mvc.Core.Test/ControllerBaseTest.cs [hunk 1]
```diff
@@ -765,7 +765,7 @@ public void HttpUnauthorized_SetsStatusCode()
             var controller = new TestableController();
 
             // Act
-            var result = controller.HttpUnauthorized();
+            var result = controller.Unauthorized();
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/ControllerBaseTest.cs]: Should change all test names that mention `HttpBlah` as well.

File: test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ControllerUnitTestabilityTests.cs [hunk 2]
```diff
@@ -688,22 +688,22 @@ public IActionResult CreatedAtRoute_Action(string routeName, object routeValues,
 
             public IActionResult HttpBadRequest_Action()
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ControllerUnitTestabilityTests.cs]: Suggest correcting these action names (and the URIs in the test class) too.

**Comment by dougbu** [comment]
:watch:

File: test/Microsoft.AspNetCore.Mvc.Core.Test/ControllerBaseTest.cs (hunk 1 above)
Comment (Eilon) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/ControllerBaseTest.cs]: Not too concerned about these. I think the test names / URLs are pretty clear either way, no?

Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/ControllerBaseTest.cs]: In other renaming PRs, we've updated test names et cetera to match the new method names. But :ok: if you're fine w/ the inconsistencies.

File: test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ControllerUnitTestabilityTests.cs (hunk 2 above)
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ControllerUnitTestabilityTests.cs]: Suspect @Eilon wants you to ignore me here (see https://github.com/aspnet/Mvc/pull/4107#discussion-diff-53069339).

File: test/Microsoft.AspNetCore.Mvc.Core.Test/ControllerBaseTest.cs (hunk 1 above)
Comment (Eilon) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/ControllerBaseTest.cs]: The goal is clarity and maintainability.

**Comment by dougbu** [comment]
Was about to say: :shipit: but use #4095 for the `ControllerBase` changes.

> The goal is clarity and maintainability.

Sounds good.

But I'm confused by renaming the result classes. Didn't @Eilon say _not_ to do that?

File: src/Microsoft.AspNetCore.Mvc.Core/StatusCodeResult.cs [hunk 3]
```diff
@@ -12,14 +12,14 @@ namespace Microsoft.AspNetCore.Mvc
     /// Represents an <see cref="ActionResult"/> that when executed will
     /// produce an HTTP response with the given response status code.
     /// </summary>
-    public class HttpStatusCodeResult : ActionResult
+    public class StatusCodeResult : ActionResult
```
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.Core/StatusCodeResult.cs]: This one isn't like the others. @Eilon does it need to change?

Comment (kichalla) [inline src/Microsoft.AspNetCore.Mvc.Core/StatusCodeResult.cs]: @Eilon approved on this change offline

Comment (Eilon) [inline src/Microsoft.AspNetCore.Mvc.Core/StatusCodeResult.cs]: Yes this was discussed and spec'ed and approved.

Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.Core/StatusCodeResult.cs]: :+1: thx

**Comment by dougbu** [comment]
:shipit: but use #4095 for the initial `ControllerBase` changes. Add the rest in a second commit.

**Comment by kichalla** [comment]
fd3ee499872465ccc2c5534b58f8ad113a8c0190

**Pull request closed without merging**